### PR TITLE
Feature/ts 30.refactoring of alerts

### PR DESCRIPTION
**Pull request by helmy-er**

### CODE REVIEW

File: app/javascript/components/Alert.tsx [hunk 1]
```diff
@@ -30,7 +30,7 @@ const SiteAlerts: React.FC<SiteAlertProps> = (props: SiteAlertProps) => {
       }}
       onClose={handleClose}
     >
-      <SnackbarAlert severity={alertType} onClose={handleClose}>{alertText}</SnackbarAlert>
+      <SnackbarAlert severity={alertData.alertType} onClose={handleClose}>{alertData.alertText}</SnackbarAlert>
```
Comment (MonkeyKingBar-bit) [inline app/javascript/components/Alert.tsx:33]: U need use obj & set data here

File: app/javascript/components/Company.tsx [hunk 2]
```diff
@@ -14,10 +14,9 @@ const Company: React.FC<CompanyProps> = (props: CompanyProps) => {
   const [isActiveModal, setModalActive] = useState(false);
   const [companies, setCompany] = React.useState(null);
   const [formErrors, setFormErrors] = React.useState([]);
-  const [alertOpen, alertSetOpen] = React.useState(false);
-  const [alertType, setAlertType] = React.useState();
-  const [alertText, setAlertText] = React.useState();
   const [searchData, setSearchData] = React.useState();
+  const [alertOpen, alertSetOpen] = React.useState<boolean>(false);
```
Comment (MonkeyKingBar-bit) [inline app/javascript/components/Company.tsx:18]: U need create only one state with obj type

**Review by MonkeyKingBar-bit — CHANGES_REQUESTED** [review]

**Comment by NDragneelL9** [comment]
Agree with Kayta

File: app/javascript/common/interfaces_types.ts [hunk 3]
```diff
@@ -240,10 +227,8 @@ export interface WaybillProps {
 }
 
 export interface SiteAlertProps {
-    alertType: any;
-    alertText: any;
-    alertOpen: boolean,
-    alertSetOpen: any;
+    alertData:any,
```
Comment (MonkeyKingBar-bit) [inline app/javascript/common/interfaces_types.ts:230]: U need need use only one var here!

File: app/javascript/components/Alert.tsx [hunk 4]
```diff
@@ -10,27 +10,27 @@ const SnackbarAlert = React.forwardRef<HTMLDivElement, AlertProps>(
...
-    alertType, alertText, alertSetOpen, alertOpen,
+    alertData, setAlertData,
   } = props;
 
   const handleClose = (event?: React.SyntheticEvent | Event, reason?: string) => {
     if (reason === 'clickaway') {
       return;
     }
-    alertSetOpen(false);
+    setAlertData({ alertType: alertData.alertType, alertText: alertData.alertText, open: false });
```
Comment (MonkeyKingBar-bit) [inline app/javascript/components/Alert.tsx:20]: why u use this?

File: app/javascript/components/Company.tsx [hunk 5]
```diff
@@ -67,16 +63,9 @@ const Company: React.FC<CompanyProps> = (props: CompanyProps) => {
...
-        setAlertText={setAlertText}
-      />
-      <SiteAlerts
-        alertType={alertType}
-        alertText={alertText}
-        alertOpen={alertOpen}
-        alertSetOpen={alertSetOpen}
+        setAlertData={setAlertData}
       />
+      <SiteAlerts alertData={alertData} setAlertData={setAlertData} />
```
Comment (MonkeyKingBar-bit) [inline app/javascript/components/Company.tsx:68]: U need use only one var!

**Review by MonkeyKingBar-bit — CHANGES_REQUESTED** [review]

**Review by MonkeyKingBar-bit — APPROVED** [review]

**Review by NDragneelL9 — APPROVED** [review]

**Pull request merged**